### PR TITLE
RAI-15329 - remove axios entirely

### PR DESCRIPTION
**Pull request by clayton-knapp**
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/reclaim-ai/src/hooks/useTask.tsx [hunk 1]
```diff
@@ -73,29 +72,38 @@ export const useTaskActions = () => {
...
     try {
-      const [task, error] = await axiosPromiseData(fetcher(`/planner/start/task/${id}`, { method: "POST" }));
-      if (!task || error) throw error;
-      return task;
+      const [intermediateResult, error] = await nodeFetchPromiseData<PlannerActionIntermediateResult>(
+        fetcher(`/planner/start/task/${id}`, { method: "POST" })
+      );
+      if (!intermediateResult || error) throw error;
+      await showHUD("Started Task");
+      return intermediateResult;
```
Comment (ScottFSeely) [inline extensions/reclaim-ai/src/hooks/useTask.tsx:80]: Seems like this logic could get consolidated across all these requests.

**Review by SandyGifford — COMMENTED** [review]
Agree with what @ScottFSeely said - although possibly in a new PR:

- do we need `nodeFetchPromiseData`?  Or can it be built in to `fetcher`
- that `JSON.stringify` can probably also go into `fetcher`
- `if (!intermediateResult || error) throw error;` could throw `undefined`

File: extensions/reclaim-ai/src/hooks/useTask.tsx (hunk 1 above)
Comment (clayton-knapp) [reply extensions/reclaim-ai/src/hooks/useTask.tsx]: done

**Comment by clayton-knapp** [comment]
> Agree with what @ScottFSeely said - although possibly in a new PR:
> 
> * do we need `nodeFetchPromiseData`?  Or can it be built in to `fetcher`
> * that `JSON.stringify` can probably also go into `fetcher`
> * `if (!intermediateResult || error) throw error;` could throw `undefined`

addressed

**Review by SandyGifford — APPROVED** [review]

**Pull request merged**